### DISCUSSION TITLE
Treating warnings as errors because process.env.CI = true

### DISCUSSION THREAD

**Discussion by simerlec** [community/community#25228] [Actions]
<p>Hey,</p>
<p>since about 2 days now our working github action fails with “Treating warnings as errors because process.env.CI = true.” because the webpack build returns a few size warnings. I tried to simply circumvent this for now by setting CI to false but I cannot figure out how. It does not work if I set it inside the workflow via</p>
<pre><code>- name: npm test:ci
      run: npm run test:ci
      env:
         CI: false
</code></pre>
<p>or if I include it in the npm command directly.</p>
<p>What am I doing wrong?</p>
<p>kind regards</p>

**Comment by ericsciple** [comment, marked as answer]
<p>can you try setting </p>
<pre><code>CI: ""
</code></pre>

**Reply by JadeShi-1018** [reply]
very helpful! Thanks a lot!

**Comment by simerlec** [comment]
<p>Hey, and thanks for the suggestion.</p>
<p>Unfortunately it did not resolve the situation. Just to be sure I added the CI env variable to the job as well as to the step in the workflow file. I also printed the <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions" rel="nofollow noopener">env object</a> to see what was in there as part of the job.</p>
<pre><code>- name: Dump GitHub env object
        env:
          ENV_CONTEXT: ${{ toJson(env) }}
        run: echo "$ENV_CONTEXT"
</code></pre>
<p>This is what I got:</p>
<pre><code>{
  "CI": ""
}
</code></pre>
<p>I must be missing something obvious…</p>

**Comment by simerlec** [comment]
<p>Wait, I think it actually worked. Huh, that’s interesting - why is empty string working but false is not?</p>
<p>Thanks a lot, <a class="mention" href="/ericsciple">@ericsciple</a> </p>

**Comment by fooey** [comment]
<p>Glad someone else ran into this, I thought I was going crazy when all my deploys quit working.</p>

**Comment by ericsciple** [comment]
<p>My best guess is the downstream tool may check for the existence of the variable (not empty) or might check for “0” instead of “false” or something</p>

**Comment by fooey** [comment]
<p>Seems to be exactly the issue</p>
<p><a href="https://github.com/facebook/create-react-app/blob/1cbc6f7db62f78747cb6ca41450099181139325e/packages/react-scripts/scripts/build.js#L198-L211" rel="nofollow noopener">https://github.com/facebook/create-react-app/blob/1cbc6f7db62f78747cb6ca41450099181139325e/packages/react-scripts/scripts/build.js#L198-L211</a> </p>
<pre><code>if (
        process.env.CI &amp;&amp;
        (typeof process.env.CI !== 'string' ||
          process.env.CI.toLowerCase() !== 'false') &amp;&amp;
        messages.warnings.length
      ) {
        console.log(
          chalk.yellow(
            '\nTreating warnings as errors because process.env.CI = true.\n' +
              'Most CI servers set it automatically.\n'
          )
        );
        return reject(new Error(messages.warnings.join('\n\n')));
      }
</code></pre>

**Comment by jonenst** [comment]
<p>So what changed ? Did github start setting the CI variable ? is there an annoucement ? It’s not mentionned in <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables#default-environment-variables" rel="nofollow noopener">https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables#default-environment-variables</a></p>
<p>Cheers,</p>
<p>Jon</p>

**Comment by jonenst** [comment]
<p>Regarding the “false” value, maybe OP had old react-scripts ? Fixed in 2017: <a href="https://github.com/facebook/create-react-app/commit/5b38c5439d9f078cbb65e3f9c6ae86cb6f9d7097" rel="nofollow noopener">https://github.com/facebook/create-react-app/commit/5b38c5439d9f078cbb65e3f9c6ae86cb6f9d7097</a></p>
<p>With recent react-scripts, using the following works:</p>
<pre><code>env:
  CI: false
</code></pre>

**Reply by colinbut** [reply]
thanks a lot for this. This now works

**Comment by itbeginswithi** [comment]
<p>I ran into the same issue while using  react@17.0.2 and react-scripts@5.0.0 when I updated my git repo, the solution for me was by following the same suggestion the original poster encountered and setting the environment CI = true in Netlify’s Environment variables.</p>

**Comment by abid000007** [comment]
change it to   CI=false npm run build
 and it will work for sure

**Reply by Vlast001** [reply]
'CI' is not recognized as an internal or external command,
operable program or batch file.

**Comment by innocent-george-mithu-nyamusa** [comment]
In your deploy script before build add this :

```
- name: Disable CI checks
        run: echo "CI=false" >> $GITHUB_ENV
```

this add CI to false in your env variables during current build.
